### DISCUSSION TITLE
How can I build and package a packaged single-project MSIX app in command line?

### DISCUSSION THREAD

**Discussion by HEIGE-PCloud** [microsoft/WindowsAppSDK#1835] [Q&A] [answered]
After migrating my existing packaged WinUI C# project to the [single-project MSIX app](https://docs.microsoft.com/en-gb/windows/apps/windows-app-sdk/single-project-msix?tabs=csharp) and getting rid of the WAP project, I find the CI failed to build a MSIX package. 

The CI uses the following command to build the app.

```powershell
 msbuild $env:Solution_Name /restore /p:Platform=$env:TargetPlatform /p:Configuration=$env:Configuration /p:UapAppxPackageBuildMode=$env:BuildMode /p:AppxBundle=$env:AppxBundle /p:PackageCertificateKeyFile=$env:PfxPath /p:PackageCertificatePassword="${{ secrets.Pfx_Key }}"
```

which works fine when the WAP project is there.

When I execute the command in my local environment, the app does get build, but no MSIX package is generated. When I use the Create App Package function in Visual Studio to package the app, an `AppPackages` folder is created correctly with the MSIX package, certificate and install Powershell script inside - which is what I expected.

I cannot find any documentation on this topic. How can I package the app in command line and fix the CI/CD pipeline?

**Comment by andrewleader** [comment, marked as answer]
Hey @HEIGE-PCloud, thanks for asking this question! This is something we need to get added to our docs (another customer asked this last week). In the meantime, I'll provide the answer here 😊

**The key missing build command option is `/p:GenerateAppxPackageOnBuild=true`**... that'll cause the MSIX package to be generated!

Here's an example of a GitHub action building a WinUI 3 single project app: https://github.com/andrewleader/WindowsAppSDKGallery/blob/main/.github/workflows/dotnet-desktop.yml#L102

Also, one other thing to note...

* Single-project MSIX currently doesn't support producing MSIX Bundles, it only produces a single MSIX. You can bundle them into a MSIX bundle using this GitHub action: https://github.com/marketplace/actions/msix-bundler
